src/app/server/Mdns.cpp

```diff
@@ -93,30 +87,33 @@ CHIP_ERROR GetCommissionableInstanceName(char * buffer, size_t bufferLen)
 /// Set MDNS operational advertisement
 CHIP_ERROR AdvertiseOperational()
 {
-    uint64_t fabricId;
-
-    if (DeviceLayer::ConfigurationMgr().GetFabricId(fabricId) != CHIP_NO_ERROR)
+    for (const Transport::FabricInfo & fabricInfo : GetGlobalFabricTable())
     {
-        ChipLogError(Discovery, "Fabric ID not known. Using a default");
-        fabricId = 5544332211;
+        if (fabricInfo.IsInitialized())
```

Comment:
This is the right thing to do here, but will break commissionable discovery for ECM and BCM because the Advertise() method has this incorrect logic:

```
    // TODO: device can re-enter commissioning mode after being fully provisioned
    // (additionalPairing == true)
    bool notYetCommissioned = !DeviceLayer::ConfigurationMgr().IsFullyProvisioned();
    bool additionalPairing  = false;
    advertiseParameters.SetCommissioningMode(notYetCommissioned, additionalPairing);
```

I think we need to remove the code in `Advertise` that is tyring to infer operational vs. commissionable advertisement.

Reply:
`IsFullyProvisioned` definitely does not report anything sensible.  Any use of it is broken code.

I guess the point is that `Advertise` can/should just use `HaveOperationalCredentials`?  I'll take a look.

Reply:
Yes, that's absolutely fine.  I was confused by this too.  It looks like we're finally in a place where we can probably do some cleanup around this though, so that's good news.

Reply:
Also, I filed https://github.com/project-chip/connectedhomeip/issues/8954 to sort out the configuration manager bits.